src/gsi/cplr_wrf_netcdf_interface.f90

```diff
@@ -1022,6 +1033,116 @@ subroutine convert_netcdf_mass_wrf(this)
           write(iunit)field2   !TH2
        endif
 
+!      Reading Significant Wave Height (HOWV) in firstguess
+       if(i_howv_3dda >= 1) then                   ! if howv is found in anavinfo
+          rmse_var='HOWV'
+          call ext_ncd_get_var_info (dh1,trim(rmse_var),ndim1,ordering,staggering, &
+               start_index,end_index, WrfType, ierr    )
+          if(print_verbose)then
+             write(6,*)' rmse_var=',trim(rmse_var)
+             write(6,*)' ordering=',ordering
+             write(6,*)' WrfType,ierr=',WrfType,ierr
+             write(6,*)' ndim1=',ndim1
+             write(6,*)' staggering=',staggering
+             write(6,*)' start_index=',start_index
+             write(6,*)' end_index=',end_index
+          end if
+          if(ierr == 0) then
+             call ext_ncd_read_field(dh1,DateStr1,TRIM(rmse_var),              &
+                  field2,WRF_REAL,0,0,0,ordering,           &
+                  staggering, dimnames ,               &
+                  start_index,end_index,               & !dom
+                  start_index,end_index,               & !mem
+                  start_index,end_index,               & !pat
+                  ierr                                 )
+             if(print_verbose)write(6,*)' max,min ',trim(adjustl(rmse_var)),'=',maxval(field2),minval(field2)
+             write(iunit)field2   !HOWV  (2D wave height)
+             i_howv_3dda = 2               ! howv was found both in anavinfo and firstguess
+          else
+             derr_msg='Warning: Error when get info in firstguess for var : '//trim(rmse_var)//    &
+                      ' ==> Stop GSI analysis ...  ierr='
+!            write(6,'(1x,A,A2,A,1x,I6)')trim(adjustl(myname_)),'::',trim(adjustl(derr_msg)),ierr
+!            call stop2(ierr)
+             call die(trim(adjustl(myname_)),trim(adjustl(derr_msg)),ierr)
+             i_howv_3dda = 0               ! skipping analysis of this variable
```

Comment:
Question:  Doesn't `die` execute `mpi_abort` and terminate program execution?  If so,  execution will not reach the `i_howv_3dda = 0` line, right?   

Comment:  The commented out line `!      call stop2(ierr)` can be removed.

Reply:
@RussTreadon-NOAA Hi Russ,
In mpeu_util.F90, "die" invokes subroutine "dropdead_", then in "dropdead_", it calls "mpi_abort" to terminate the GSI code (ifdef USE_MPI_ABORT, and it is defined), otherwise it calls "exit" to stop the running of GSI. 
Yes, here "die" terminates the execution. 

Yes, your right. Since the GSI code is going to be terminated under this condition, so it is not necessary to update "i_howv_3dda". The reason that I still keep this line here, is that if in the future it is suggested to keep the code running, instead of calling die, then the updated "i_howv_3dda" is ready to be broadcast to all PEs. So I will move the line "i_howv_3dda = 0" before call die in the updated new code.

Thank you!